addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -593,5 +626,63 @@ odoo.define('website.s_website_form', function (require) {
         _onFieldInput() {
             this._updateFieldsVisibility();
         },
+        /**
+         * Detects if files are uploaded, in order to modify the text displayed
+         * on the input file button afterwards.
+         * The modification is made by hiding the real button and then creating
+         * a fake one, along with a text displaying information about the
+         * uploaded files next to it.
+         * @private
+         * @param {MouseEvent} ev
+         */
+        _onFileChange(ev) {
+            const realButton = ev.currentTarget;
+            const parentNode = ev.target.parentNode;
+
+            // Create a fake button with "Replace Files" as text.
+            if (!parentNode.querySelector('input[class="fake-button"]')) {
+                // Hide the current input button.
+                realButton.setAttribute('style', 'display:none');
+
+                // Create the fake button.
+                const fakeButton = document.createElement('input');
+                fakeButton.type = 'button';
+                fakeButton.className += 'fake-button';
+                fakeButton.value = 'Replace Files';
+                fakeButton.title = realButton.title;
+                parentNode.appendChild(fakeButton);
+
+                // Create the information text (empty for now).
+                const fileInfo = document.createElement('label');
+                fileInfo.className += 'file-info';
+                fileInfo.style = 'font-weight:normal'; // To remove the bold effect.
+                parentNode.appendChild(fileInfo);
+            }
+
+            const fileInfo = parentNode.querySelector('label[class="file-info"]');
+            const fileNumber = realButton.files.length;
+            if (fileNumber === 0) {
+                // If no files were uploaded, put the real button back.
+                const fakeButton = parentNode.querySelector('input[class="fake-button"]');
+                parentNode.removeChild(fakeButton);
+                parentNode.removeChild(fileInfo);
+                realButton.removeAttribute('style');
+            } else {
+                // Update the information text.
+                const text = fileNumber === 1 ?
+                    realButton.files[0].name : fileNumber + ' files selected';
+                fileInfo.innerHTML = text;
```

Comment:
Sounds unsafe, it might allow for HTML injection. If you use `fileInfo.innerText` any string will be considered as text, and therefore be escaped adequately.